tests/unit/src/test/scala/tests/FormattingSlowSuite.scala

```diff
@@ -225,4 +229,99 @@ object FormattingSlowSuite extends BaseSlowSuite("formatting") {
     } yield ()
   }
 
+  testAsync("missing-version") {
+    cleanWorkspace()
+    client.showMessageRequestHandler = { params =>
+      if (MissingScalafmtVersion.isMissingScalafmtVersion(params)) {
+        params.getActions.asScala
+          .find(_ == MissingScalafmtVersion.changeVersion)
+      } else {
+        None
+      }
+    }
+    for {
+      _ <- server.initialize(
+        """|/.scalafmt.conf
+           |maxColumn=40
+           |/Main.scala
+           |object   Main
+           |""".stripMargin,
+        expectError = true
+      )
+      _ <- server.didOpen("Main.scala")
+      _ <- server.formatting("Main.scala")
```

Comment:
To explain the problem, this future here returned before we were able to fix `.scalafmt.conf`. Now the `formatting` future includes waiting for the "update .scalafmt.conf?" dialogue.